addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -1057,6 +1057,7 @@ export class PosStore extends Reactive {
                 // a higher id than the original one since it's the last one created.
                 const session = this.models["pos.session"].sort((a, b) => a.id - b.id)[0];
                 session.delete();
+                this.models["pos.order"].getAll()[0].session_id = this.session;
```

Comment:
It's indeed better BUT you still only change the first order and not all.
```suggestion
                this.models["pos.order"].getAll().forEach((order) => order.session_id = this.session);
```
Something like this should do the trick. After that we'll be good :+1: